src/coreclr/System.Private.CoreLib/src/System/GC.CoreCLR.cs

```diff
@@ -800,7 +800,9 @@ public static unsafe T[] AllocateUninitializedArray<T>(int length, bool pinned =
             if (pinned)
                 flags |= GC_ALLOC_FLAGS.GC_ALLOC_PINNED_OBJECT_HEAP;
 
-            return Unsafe.As<T[]>(AllocateNewArray(RuntimeTypeHandle.ToIntPtr(typeof(T[]).TypeHandle), length, flags));
+            T[]? result = null;
+            AllocateNewArray(TypeHandle.TypeHandleOf<T[]>().AsMethodTable(), length, flags, ObjectHandleOnStack.Create(ref result));
```

Comment:
`RuntimeTypeHandle.ToIntPtr(typeof(T[]).TypeHandle` is recognized as a batch of JIT intrinsics and it will generate MethodTable* as a constant in the final code. I think `AsMethodTable` will be a perf regression.

Reply:
I ran it locally and didn't see any regression. I can revert back to `RuntimeTypeHandle.ToIntPtr(typeof(T[]).TypeHandle` if that is preferred.

Reply:
I have checked the disassembly and I see a small regression in shared generic code (~3 extra instructions) that is a lot less than I have expected and it is not something to worry about.  Feel free to switch it to back to `TypeHandle.TypeHandleOf<T[]>` if you prefer.